pkg/resource/function/hooks.go

```diff
@@ -476,35 +476,44 @@ func customPreCompare(
 	if ackcompare.HasNilDifference(a.ko.Spec.Code, b.ko.Spec.Code) {
 		delta.Add("Spec.Code", a.ko.Spec.Code, b.ko.Spec.Code)
 	} else if a.ko.Spec.Code != nil && b.ko.Spec.Code != nil {
```

Comment:
Should we consider removing this customPreCompare, generate all this code, and then rely on the custom update function to lookup `*a.ko.Spec.PackageType == "Image"|"Zip"`